solutions/CF They are Everywhere.mdx

```diff
@@ -0,0 +1,181 @@
+---
+id: cf-701C
+source: CF
+title: They are Everywhere
+author: Kevin Sheng
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/46283)
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+ 
+using std::cout;
+using std::endl;
+using std::vector;
+ 
+/**
+ * https://codeforces.com/problemset/problem/701/C
+ * 7
+ * bcAAcbc should output 3
+ * 6
+ * aaBCCe should output 5
+ */
+int main() {
+    int flat_num;
+    std::cin >> flat_num;
+    vector<char> flats(flat_num);
+    std::unordered_set<char> types;
+    for (char& p : flats) {
+        std::cin >> p;
+        types.insert(p);
+    }
+ 
+    int lo = 0;
+    int hi = flat_num;
+    int valid = -1;
+    while (lo <= hi) {
+        int mid = (lo + hi) / 2;
+ 
+        bool all_catchable = false;    
+        std::unordered_map<char, int> curr_pokemon;
+        for (int to_add = 0; to_add < flat_num; to_add++) {
+            curr_pokemon[flats[to_add]]++;
+            if (to_add >= mid) {
+                char to_remove = flats[to_add - mid];
+                curr_pokemon[to_remove]--;
+                if (curr_pokemon[to_remove] == 0) {
+                    curr_pokemon.erase(to_remove);
+                }
+            }
+            if (curr_pokemon.size() == types.size()) {
+                all_catchable = true;
+                break;
+            }
+        }
+    
+        if (all_catchable) {
+            valid = mid;
+            hi = mid - 1;
+        } else {
+            lo = mid + 1;
+        }
```

Comment:
No need to binary search. You can use 2-pointer for this (extend if not all pokemon are present, contract if all pokemon are present)